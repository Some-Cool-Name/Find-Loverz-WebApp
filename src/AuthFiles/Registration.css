body {
    margin: 0;
    font-family: Arial, Helvetica, sans-serif;
}

.header{
    height: 45px;
    overflow: hidden;
    background:linear-gradient(to right,#c471ed,#b621fe , #12c2e9); 
}
label.fieldDescription{
    font-weight: bold;
    box-sizing: border-box;
    text-align: start;
    width: 100%;
    height: fit-content;
    padding: 1px;
    padding-left: 10px;
    padding-bottom: 1px;
    margin-inline-start: auto;
    margin-inline-end: auto;
}
div {
    display: block;
}

.header{
    height: 45px;
    overflow: hidden;
    background:linear-gradient(to right,#c471ed,#b621fe , #12c2e9); 
}
input.inputbox{
    margin-top: 2px;
}
h3{
  margin-top: auto;
  margin-bottom: opx;
    align-items: baseline;
    color: white;
    font-family:cursive;
    font-size: xx-large;
}

.form{
    width: 350px;
    height: fit-content;

    flex-direction: column;
    
    padding: 20px 55px 20px 55px;
    margin: 60px auto;
    border-radius: 25px;

    background: #ffffff;
    box-shadow: 0px 14px 80px rgba(34, 35, 58, 0.2);
    
    transition: all .3s;
}
hr{
    margin-top: 0px;
    margin-bottom: 20px;
    background: linear-gradient(to right,#c471ed,#b621fe , #12c2e9);
    margin-left: auto;
    margin-right: auto;
}
.input-boxes{
    padding: 0%;
}

label.intrestDescription{
    font-weight: bold;
    box-sizing: border-box;
    width: 100%;
    padding: 1px;
    padding-left: 15px;
    padding-bottom: 1px;
    margin-inline-start: auto;
    margin-inline-end: auto;
    margin-right: 150px;
    align-items: center;
}
input,
select{
    box-sizing: border-box;
    border: 1px solid #ccc;
    border-radius: 1.5em;
    background: transparent;
    resize: none;
    outline: none;
    top: 0;
    left: 0;   
}

select{
    -webkit-appearance: none;
    -moz-appearance: none;
}

input:focus{
    border-color:#b621fe;
}
.button{
    cursor: pointer;
    background: linear-gradient(to right,#c471ed,#b621fe , #12c2e9);
    box-shadow: 0px 5px 15px 5px #840fe440;
    color: #fff;
    padding: 10px;
    font-size: 1em;
    font-weight: bold;
    width: 100%;
    border-radius: 25px;
    border: none;
    height: 25px;
    width: 140px;
    margin: 20px 0px;
}
.intrestbutton,
#register-button,
#interests-close{
    min-width:  200px;
    width: fit-content;
    height: fit-content;

    cursor: pointer;

    color: #fff;
    background: linear-gradient(to right,#c471ed,#b621fe , #12c2e9);
    box-shadow: 0px 5px 15px 5px #840fe440;
    
    padding: 10px;
    margin: 20px 0px;
    border-radius: 25px;
    border: none;

    font-size: 1em;
    text-transform: uppercase;
    font-weight: bold;
    align-items: center;
    transition: ease-in-out 0.5s;
}

link.button{
    appearance: auto;
    user-select: none;
    white-space: pre;
    align-items: center;
    box-sizing: border-box;
    width: 100%;
   
}

#interests-close{
    width: fit-content;
    min-width: none;
}

input#fileupload{
    border: none;
    border-radius: 0;
    padding: 0;
}

input#fileupload::-webkit-file-upload-button {
    visibility: hidden;
}

input#fileupload::before {
    content: 'Choose File';
    display: inline-block;

    border: none;
    border-radius: 8px;
    padding: 10px;

    font-weight: 700;
    font-size: 10pt;
    text-transform: uppercase;

    color: #ffffff;
    background: linear-gradient(to right,#c471ed,#b621fe , #12c2e9);
    
    cursor: pointer;
    outline: none;
    white-space: nowrap;
    -webkit-user-select: none; 
}

#register-pic{
    max-width: 320px;
    max-height: 320px;
    
    border-radius: 20px;
}