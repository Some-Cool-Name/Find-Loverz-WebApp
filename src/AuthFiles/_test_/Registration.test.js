import React from 'react';
import { render, screen, waitFor, fireEvent, waitForElementToBeRemoved } from '@testing-library/react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import { BrowserRouter } from "react-router-dom"
import Registration from '../Registration';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

const server = setupServer(
    rest.get('https://lamp.ms.wits.ac.za/home/s1851427/webb.php', (req, res, ctx) => {
      // Respond with a mocked user token that gets persisted
      // in the `sessionStorage` by the `Login` component.
      console.log("hello 2");
      return res( ctx.status(200), ctx.json(
          {"login":[
              {"username":"demo",
              "name":"demo",
              "Birthday":"10-05-1993",
              "gender":"Male",
              "Sexuality":"Female"
            }],
            "success":"1",
            "message":"success"}))
    }),
  )
  
  // Enable API mocking before tests.
  beforeAll(() => server.listen())
  
  // Reset any runtime request handlers we may add during the tests.
  afterEach(() => server.resetHandlers())
  
  // Disable API mocking after the tests are done.
  afterAll(() => server.close())

it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter>
        <Registration/>
    </BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Registration Inputs : Empty Fields Testing , Should return error', async () => {
    // RENDER OR LOAD THE REGISTRATION PAGE
    render(
        <BrowserRouter>
            <Registration/>
        </BrowserRouter>
    );
    // ACCESS THE USERNAME FIELD ON THE REGISTRATION PAGE
    const userNameLabel = screen.getByLabelText('Username'); 
    fireEvent.change(userNameLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(userNameLabel).toHaveValue('');
    });

    // ACCESS THE PASSWORD FIELD ON THE REGISTRATION PAGE
    const passwordLabel = screen.getByLabelText('Password');
    fireEvent.change(passwordLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(passwordLabel).toHaveValue('');
    });

    // ACCESS THE CONFIRM PASSWORD FIELD ON THE REGISTRATION PAGE
    const confirmPaswordLabel = screen.getByLabelText('Confirm password');
    fireEvent.change(confirmPaswordLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(confirmPaswordLabel).toHaveValue('');
    });

    // ACCESS THE NAME FIELD ON THE REGISTRATION PAGE
    const nameLabel = screen.getByLabelText('Name');
    fireEvent.change(nameLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(nameLabel).toHaveValue('');
    });

    // THESE ARE DROPDOWNS : ACCESS THE GENDER DROPDOWN 
    const genderLabel = screen.getByLabelText('Gender');
    expect(genderLabel).toBeTruthy();
    // SELECT THE FEMALE OPTION ON THE GENDER DROPDOWN
    userEvent.selectOptions(genderLabel, 'Female');

    // ACCESS THE BIRTHDAY FIELD ON THE REGISTRATION PAGE
    const birthdayLabel = screen.getByLabelText('Birthday');
    fireEvent.change(birthdayLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(birthdayLabel).toHaveValue('');
    });

    // THESE ARE DROPDOWNS : ACCESS THE SEXUALITY DROPDOWN 
    const sexualityLabel = screen.getByLabelText('Sexuality');
    expect(sexualityLabel).toBeTruthy();
    // SELECT THE MALE OPTION ON THE SEXUALITY DROPDOWN
    userEvent.selectOptions(sexualityLabel, 'Male');

    // ACCESS THE LOCATION FIELD ON THE REGISTRATION PAGE
    const locationLabel = screen.getByLabelText('Location');
    fireEvent.change(locationLabel, {target: {value: ''}}) // type an empty string
    await waitFor(() => { 
        expect(locationLabel).toHaveValue('');
    });

    // ACCESS THE REGISTER BUTTON ON THE REGISTRATION PAGE
    const registerBtn = screen.getByDisplayValue('Register');
    fireEvent.click(registerBtn);

    await waitFor(() => { 
        expect(registerBtn).toBeTruthy();
    });

    // you expect something from the database to happen


});

it('Registration Inputs : Field Filled correctly, but user already exists , Should return user exists error', async () => {
    // RENDER OR LOAD THE REGISTRATION PAGE
    server.use(
        rest.get('https://lamp.ms.wits.ac.za/home/s1851427/webb.php', (req, res, ctx) => {
          // Respond with "200 Internal Server Error" status for this test.
          return res(
            ctx.status(200),
            ctx.json('user exists'),
          );
        }),
      );

    render(
        <BrowserRouter>
            <Registration/>
        </BrowserRouter>
    );
    // ACCESS THE USERNAME FIELD ON THE REGISTRATION PAGE
    const userNameLabel = screen.getByLabelText('Username'); 
    fireEvent.change(userNameLabel, {target: {value: 'demo'}}) // type username
    await waitFor(() => { 
        expect(userNameLabel).toHaveValue('demo');
    });

    // ACCESS THE PASSWORD FIELD ON THE REGISTRATION PAGE
    const passwordLabel = screen.getByLabelText('Password');
    fireEvent.change(passwordLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(passwordLabel).toHaveValue('demo');
    });

    // ACCESS THE CONFIRM PASSWORD FIELD ON THE REGISTRATION PAGE
    const confirmPaswordLabel = screen.getByLabelText('Confirm password');
    fireEvent.change(confirmPaswordLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(confirmPaswordLabel).toHaveValue('demo');
    });

    // ACCESS THE CONFIRM PASSWORD FIELD ON THE REGISTRATION PAGE
    const nameLabel = screen.getByLabelText('Name');
    fireEvent.change(nameLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(nameLabel).toHaveValue('demo');
    });

    // THESE ARE DROPDOWNS : ACCESS THE GENDER DROPDOWN 
    const genderLabel = screen.getByLabelText('Gender');
    expect(genderLabel).toBeTruthy();
    // SELECT THE FEMALE OPTION ON THE GENDER DROPDOWN
    userEvent.selectOptions(genderLabel, 'Female');

    // ACCESS THE BIRTHDAY FIELD ON THE REGISTRATION PAGE
    const birthdayLabel = screen.getByLabelText('Birthday');
    fireEvent.change(birthdayLabel, {target: {value: '2000-01-01'}}) // type an empty string
    await waitFor(() => { 
        expect(birthdayLabel).toHaveValue('2000-01-01');
    });

    // THESE ARE DROPDOWNS : ACCESS THE SEXUALITY DROPDOWN 
    const sexualityLabel = screen.getByLabelText('Sexuality');
    expect(sexualityLabel).toBeTruthy();
    // SELECT THE MALE OPTION ON THE SEXUALITY DROPDOWN
    userEvent.selectOptions(sexualityLabel, 'Male');

    // ACCESS THE LOCATION FIELD ON THE REGISTRATION PAGE
    const locationLabel = screen.getByLabelText('Location');
    fireEvent.change(locationLabel, {target: {value: 'Jhb'}}) // type an empty string
    await waitFor(() => { 
        expect(locationLabel).toHaveValue('Jhb');
    });

    // ACCESS THE REGISTER BUTTON ON THE REGISTRATION PAGE
    const registerBtn = screen.getByDisplayValue('Register');
    fireEvent.click(registerBtn);

    await waitFor(() => { 
        expect(registerBtn).toBeTruthy();
    });

    // you expect something from the database to happen


});

it('Registration Inputs : Field Filled correctly, user does not exists , Should return success', async () => {
    // RENDER OR LOAD THE REGISTRATION PAGE
    server.use(
        rest.get('https://lamp.ms.wits.ac.za/home/s1851427/webb.php', (req, res, ctx) => {
          // Respond with "400 Internal Server Error" status for this test.
          console.log("user registered successfully");
          return res(
            ctx.status(200),
            ctx.json('success'),
          );
        }),
      );
    
    render(
        <BrowserRouter>
            <Registration/>
        </BrowserRouter>
    );
    // ACCESS THE USERNAME FIELD ON THE REGISTRATION PAGE
    const userNameLabel = screen.getByLabelText('Username'); 
    fireEvent.change(userNameLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(userNameLabel).toHaveValue('demo');
    });

    // ACCESS THE PASSWORD FIELD ON THE REGISTRATION PAGE
    const passwordLabel = screen.getByLabelText('Password');
    fireEvent.change(passwordLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(passwordLabel).toHaveValue('demo');
    });

    // ACCESS THE CONFIRM PASSWORD FIELD ON THE REGISTRATION PAGE
    const confirmPaswordLabel = screen.getByLabelText('Confirm password');
    fireEvent.change(confirmPaswordLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(confirmPaswordLabel).toHaveValue('demo');
    });

    // ACCESS THE CONFIRM PASSWORD FIELD ON THE REGISTRATION PAGE
    const nameLabel = screen.getByLabelText('Name');
    fireEvent.change(nameLabel, {target: {value: 'demo'}}) // type an empty string
    await waitFor(() => { 
        expect(nameLabel).toHaveValue('demo');
    });

    // THESE ARE DROPDOWNS : ACCESS THE GENDER DROPDOWN 
    const genderLabel = screen.getByLabelText('Gender');
    expect(genderLabel).toBeTruthy();
    // SELECT THE FEMALE OPTION ON THE GENDER DROPDOWN
    userEvent.selectOptions(genderLabel, 'Female');

    // ACCESS THE BIRTHDAY FIELD ON THE REGISTRATION PAGE
    const birthdayLabel = screen.getByLabelText('Birthday');
    fireEvent.change(birthdayLabel, {target: {value: '2000-01-01'}}) // type an empty string
    await waitFor(() => { 
        expect(birthdayLabel).toHaveValue('2000-01-01');
    });

    // THESE ARE DROPDOWNS : ACCESS THE SEXUALITY DROPDOWN 
    const sexualityLabel = screen.getByLabelText('Sexuality');
    expect(sexualityLabel).toBeTruthy();
    // SELECT THE MALE OPTION ON THE SEXUALITY DROPDOWN
    userEvent.selectOptions(sexualityLabel, 'Male');

    // ACCESS THE LOCATION FIELD ON THE REGISTRATION PAGE
    const locationLabel = screen.getByLabelText('Location');
    fireEvent.change(locationLabel, {target: {value: 'Jhb'}}) // type an empty string
    await waitFor(() => { 
        expect(locationLabel).toHaveValue('Jhb');
    });

    // ACCESS THE REGISTER BUTTON ON THE REGISTRATION PAGE
    const registerBtn = screen.getByDisplayValue('Register');
    fireEvent.click(registerBtn);

    await waitFor(() => { 
        expect(registerBtn).toBeTruthy();
    });

    // you expect something from the database to happen


});
