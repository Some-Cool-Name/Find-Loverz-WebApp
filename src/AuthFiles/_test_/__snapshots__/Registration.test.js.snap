// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="registration"
>
  <h1>
    Register
  </h1>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <div
      className="form-element"
    >
      <label
        htmlFor="username"
      >
         Username
      </label>
      <input
        id="username"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="password"
      >
         Password
      </label>
      <input
        id="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="confirmPassword"
      >
         Confirm password
      </label>
      <input
        id="confirmPassword"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="name"
      >
         Name
      </label>
      <input
        id="name"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="gender"
      >
         Gender
      </label>
      <select
        id="gender"
        onChange={[Function]}
      >
        <option>
          Select an Option
        </option>
        <option>
          Male
        </option>
        <option>
          Female
        </option>
        <option>
          Other
        </option>
      </select>
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="birthday"
      >
         Birthday
      </label>
      <input
        id="birthday"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="sexuality"
      >
         Sexuality
      </label>
      <select
        id="sexuality"
        onChange={[Function]}
      >
        <option>
          Select an Option
        </option>
        <option>
          Male
        </option>
        <option>
          Female
        </option>
        <option>
          Other
        </option>
      </select>
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="interests"
      >
         Interests
      </label>
      <button
        id="interests-button"
        onClick={[Function]}
      >
        Choose Interests
      </button>
    </div>
    <div
      className="form-element"
    >
      <label
        htmlFor="location"
      >
         Location
      </label>
      <input
        id="location"
      />
    </div>
    <input
      id="register-button"
      type="submit"
      value="Register"
    />
  </form>
  <div
    id="interests-background"
  >
    <div
      id="interests-interface"
    >
      <h1>
        Choose Five Interests
      </h1>
      <div
        className="interests-container"
      >
        <div
          className="interest-element"
          id="i1"
          onClick={[Function]}
        >
          Art
        </div>
        <div
          className="interest-element"
          id="i2"
          onClick={[Function]}
        >
          Baking
        </div>
        <div
          className="interest-element"
          id="i3"
          onClick={[Function]}
        >
          Board Games
        </div>
        <div
          className="interest-element"
          id="i4"
          onClick={[Function]}
        >
          Comedy
        </div>
        <div
          className="interest-element"
          id="i5"
          onClick={[Function]}
        >
          Cooking
        </div>
        <div
          className="interest-element"
          id="i6"
          onClick={[Function]}
        >
          Cycling
        </div>
        <div
          className="interest-element"
          id="i7"
          onClick={[Function]}
        >
          Dancing
        </div>
        <div
          className="interest-element"
          id="i8"
          onClick={[Function]}
        >
          Fashion
        </div>
        <div
          className="interest-element"
          id="i9"
          onClick={[Function]}
        >
          Fishing
        </div>
        <div
          className="interest-element"
          id="i10"
          onClick={[Function]}
        >
          Foodie
        </div>
        <div
          className="interest-element"
          id="i11"
          onClick={[Function]}
        >
          Football
        </div>
        <div
          className="interest-element"
          id="i12"
          onClick={[Function]}
        >
          Gamer
        </div>
        <div
          className="interest-element"
          id="i13"
          onClick={[Function]}
        >
          Gardening
        </div>
        <div
          className="interest-element"
          id="i14"
          onClick={[Function]}
        >
          Golf
        </div>
        <div
          className="interest-element"
          id="i15"
          onClick={[Function]}
        >
          Gym
        </div>
        <div
          className="interest-element"
          id="i16"
          onClick={[Function]}
        >
          Hiking
        </div>
        <div
          className="interest-element"
          id="i17"
          onClick={[Function]}
        >
          Karaoke
        </div>
        <div
          className="interest-element"
          id="i18"
          onClick={[Function]}
        >
          Movies
        </div>
        <div
          className="interest-element"
          id="i19"
          onClick={[Function]}
        >
          Music
        </div>
        <div
          className="interest-element"
          id="i20"
          onClick={[Function]}
        >
          Netflix
        </div>
        <div
          className="interest-element"
          id="i21"
          onClick={[Function]}
        >
          Pet Lover
        </div>
        <div
          className="interest-element"
          id="i22"
          onClick={[Function]}
        >
          Photography
        </div>
        <div
          className="interest-element"
          id="i23"
          onClick={[Function]}
        >
          Picnicking
        </div>
        <div
          className="interest-element"
          id="i24"
          onClick={[Function]}
        >
          Politics
        </div>
        <div
          className="interest-element"
          id="i25"
          onClick={[Function]}
        >
          Reading
        </div>
        <div
          className="interest-element"
          id="i26"
          onClick={[Function]}
        >
          Shopping
        </div>
        <div
          className="interest-element"
          id="i27"
          onClick={[Function]}
        >
          Spirituality
        </div>
        <div
          className="interest-element"
          id="i28"
          onClick={[Function]}
        >
          Social Media
        </div>
        <div
          className="interest-element"
          id="i29"
          onClick={[Function]}
        >
          Sport
        </div>
        <div
          className="interest-element"
          id="i30"
          onClick={[Function]}
        >
          Surfing
        </div>
        <div
          className="interest-element"
          id="i31"
          onClick={[Function]}
        >
          Swimming
        </div>
        <div
          className="interest-element"
          id="i32"
          onClick={[Function]}
        >
          Travel
        </div>
        <div
          className="interest-element"
          id="i33"
          onClick={[Function]}
        >
          Vlogging
        </div>
        <div
          className="interest-element"
          id="i34"
          onClick={[Function]}
        >
          Wine
        </div>
        <div
          className="interest-element"
          id="i35"
          onClick={[Function]}
        >
          Writing
        </div>
        <div
          className="interest-element"
          id="i36"
          onClick={[Function]}
        >
          Yoga
        </div>
      </div>
      <div
        id="interests-button-container"
      >
        <button
          id="interests-cancel"
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          id="interests-confirm"
          onClick={[Function]}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
